MATLAB/mimo_driver.m

```diff
@@ -46,208 +48,99 @@
 
                 n_bs_sdrs = length(obj.bs_serial_ids);
                 n_ue_sdrs = length(obj.ue_serial_ids);
+                n_ref_sdrs = length(obj.ref_serial_ids);
 
                 obj.n_bs_ant = n_bs_sdrs * length(sdr_params.bs_ant);
                 obj.n_ue_ant = n_ue_sdrs * length(sdr_params.ue_ant);
                 obj.bs_channels = sdr_params.bs_ant;
                 obj.ue_channels = sdr_params.ue_ant;
 
+                % Base Station
                 bs_id_str = cell(1, n_bs_sdrs);
                 for i = 1:n_bs_sdrs
                     bs_id_str(1, i) = {convertStringsToChars(obj.bs_serial_ids(i))};
                 end
                 bs_id_list = py.list(bs_id_str);
 
+                % UEs
                 ue_id_str = cell(1, n_ue_sdrs);
                 for i = 1:n_ue_sdrs
                     ue_id_str(1, i) = {convertStringsToChars(obj.ue_serial_ids(i))};
                 end
                 ue_id_list = py.list(ue_id_str);
 
+                % Reference Node (overkill... it will always be at most one)
+                ref_id_str = cell(1, n_ref_sdrs);
+                for i = 1:n_ref_sdrs
+                    ref_id_str(1, i) = {convertStringsToChars(obj.ref_serial_ids(i))};
+                end
+                ref_id_list = py.list(ref_id_str);
```

Comment:
This is screaming to make a function..............

Reply:
will do it on the next commit